src/configuration-client/src/main/java/ee/ria/xroad/common/conf/globalconf/ConfigurationDownloader.java

```diff
@@ -137,6 +149,51 @@ DownloadResult download(ConfigurationSource source, String... contentIdentifiers
         return result.failure();
     }
 
+    private void supplementInternalVerificationCerts(ConfigurationLocation location)
+            throws CertificateEncodingException, IOException {
+        var sources = getAdditionalSources(location);
+        var internalVerificationCerts = sources.stream()
+                .flatMap(src -> src.getInternalVerificationCerts().stream())
+                .toList();
+        addSupplementaryVerificationCerts(location, internalVerificationCerts);
+    }
+
+    private void supplementExternalVerificationCerts(ConfigurationLocation location)
+            throws CertificateEncodingException, IOException {
+        var sources = getAdditionalSources(location);
+        var externalVerificationCerts = sources.stream()
+                .flatMap(src -> src.getExternalVerificationCerts().stream())
+                .toList();
+        addSupplementaryVerificationCerts(location, externalVerificationCerts);
+    }
+
+    private List<SharedParameters.ConfigurationSource> getAdditionalSources(ConfigurationLocation location)
+            throws CertificateEncodingException, IOException {
+        Path sharedParamsPath = fileNameProvider.getConfigurationDirectory(location.getInstanceIdentifier())
+                .resolve(ConfigurationConstants.FILE_NAME_SHARED_PARAMETERS);
+        if (sharedParamsPath.toFile().exists() && isCurrentVersion(sharedParamsPath)) {
+            SharedParameters sharedParams = new SharedParametersV3(sharedParamsPath, OffsetDateTime.MAX).getSharedParameters();
+            return sharedParams.getSources().stream()
+                    .filter(source -> location.getDownloadURL().contains(source.getAddress()))
+                    .toList();
+        }
+        return List.of();
+    }
+
+    private void addSupplementaryVerificationCerts(ConfigurationLocation location, List<byte[]> potentialCertsToBeAdded) {
+        List<byte[]> certsToBeAdded = new ArrayList<>();
+        potentialCertsToBeAdded.forEach(potentialCert -> {
+            if (location.getVerificationCerts().stream().noneMatch(cert -> Arrays.equals(cert, potentialCert))) {
+                certsToBeAdded.add(potentialCert);
+            }
+        });
+
+        if (!certsToBeAdded.isEmpty()) {
+            log.info("Adding supplementary verification certificates from shared parameters configuration file");
```

Comment:
can we move this to debug or trace?

Reply:
This line is logged when a new configuration signing key is added or when the location cache (containing the verification certificates) is invalidated, which in practice happens very rarely. Leaving it on info means that this line is logged in production by default, which might come in handy.